system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -520,11 +605,75 @@ cpr::Response ElasticStackLogAccess::performESQuery(const LogAccessConditions &
 bool ElasticStackLogAccess::fetchLog(const LogAccessConditions & options, StringBuffer & returnbuf, LogAccessLogFormat format)
 {
     cpr::Response esresp = performESQuery(options);
-    processESJsonResp(esresp, returnbuf, format);
+    processESSearchJsonResp(esresp, returnbuf, format);
 
     return true;
 }
 
+class ELASTICSTACKLOGACCESS_API ElasticStackLogStream : public CInterfaceOf<IRemoteLogAccessStream>
+{
+public:
+    virtual bool readLogEntries(StringBuffer & record, unsigned & recsRead, unsigned & totalRecs) override
+    {
+        Json::Value res;
+        recsRead = 0;
+        totalRecs = 0;
+
+        if (m_esSroller.next(res))
+        {
+
+            if (!res["total"]["value"].empty())
+            {
+                totalRecs = res["total"]["value"].asUInt();
+            }
+
+            if (!res["hits"].empty())
+            {
+                recsRead = res["hits"].size();
+                std::ostringstream sout;
+                m_jsonWriter->write(res, &sout);
+                processESScrollJsonResp(sout.str().c_str(), record, m_outputFormat);
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    ElasticStackLogStream(std::string & queryString, const char * connstr, const char * indexsearchpattern, LogAccessLogFormat format,  std::size_t pageSize, std::string scrollTO)
+     : m_esSroller(std::make_shared<elasticlient::Client>(std::vector<std::string>({connstr})), pageSize, scrollTO)
+    {
+        m_outputFormat = format;
+        m_esSroller.init(indexsearchpattern, DEFAULT_ES_DOC_TYPE, queryString);
+        m_jsonWriter = m_jsonStreamBuilder.newStreamWriter();
+    }
+
+    virtual ~ElasticStackLogStream() override
+    {
+        m_esSroller.clear();
+    }
+
+private:
+    elasticlient::Scroll m_esSroller;
+
+    LogAccessLogFormat m_outputFormat;
+    Json::StreamWriterBuilder m_jsonStreamBuilder;
+    Json::StreamWriter * m_jsonWriter;
+};
+
+IRemoteLogAccessStream * ElasticStackLogAccess::getLogReader(const LogAccessConditions & options, LogAccessLogFormat format)
+{
+    return getLogReader(options, format, DEFAULT_MAX_RECORDS_PER_FETCH);
+}
+
+IRemoteLogAccessStream * ElasticStackLogAccess::getLogReader(const LogAccessConditions & options, LogAccessLogFormat format, unsigned int pageSize)
+{
+    std::string queryString;
+    std::string queryIndex;
+    createQueryString(queryString, queryIndex, options);
+    return new ElasticStackLogStream(queryString, m_esConnectionStr.str(), m_globalIndexSearchPattern.str(), format, pageSize, DEFAULT_SCROLL_TIMEOUT);
```

Comment:
minor: Better to pass queryIndex for m_globalIndexSearchPattern?  Otherwise it is confusing why it is returned.

Reply:
great catch, fixing.